wallet/src/repository/wallets/mod.rs

```diff
@@ -116,9 +117,10 @@ impl<T: Database> Wallets<T> {
         Ok(())
     }
 
+    /// Gets a wallet's salt and IV based on its provided ID
     pub fn wallet_salt_and_iv(&self, id: &str) -> Result<(Vec<u8>, Vec<u8>)> {
-        let salt = self.db.get(&keys::wallet_id_salt(id))?;
-        let iv = self.db.get(&keys::wallet_id_iv(id))?;
+        let salt = self.db.get(&keys::wallet_id_salt(&id))?;
+        let iv = self.db.get(&keys::wallet_id_iv(&id))?;
```

Comment:
This shouldn't be needed because `id` is already a reference

```suggestion
        let salt = self.db.get(&keys::wallet_id_salt(id))?;
        let iv = self.db.get(&keys::wallet_id_iv(id))?;
```